File: LSL/liblsl-Java/examples/ReceiveDataInChunks.java
package examples;
import edu.ucsd.sccn.LSL;

public class ReceiveDataInChunks {
    public static void main(String[] args) throws Exception  {
        System.out.println("Resolving an EEG stream...");
        LSL.StreamInfo[] results = LSL.resolve_stream("type","EEG");

        // open an inlet
        LSL.StreamInlet inlet = new LSL.StreamInlet(results[0]);
        
        // receive data
        float[] sample = new float[10*inlet.info().channel_count()];
        double[] timestamps = new double[10];
        while (true) {
            inlet.pull_chunk(sample,timestamps);
            for (int i=0;i<timestamps.length;i++) {
                for (int k=0;k<sample.length;k++)
                    System.out.print("\t" + Double.toString(sample[k]));
                System.out.println();
            }
        }
    }
}

File: LSL/liblsl-Csharp/examples/ReceiveDataInChunks.cs
using System;
using System.Threading;
using LSL;

namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            // wait until an EEG stream shows up
            liblsl.StreamInfo[] results = liblsl.resolve_stream("type", "EEG");

            // open an inlet and print meta-data
            liblsl.StreamInlet inlet = new liblsl.StreamInlet(results[0]);
            System.Console.Write(inlet.info().as_xml());

            // read samples
            float[,] buffer = new float[512, 8];
            double[] timestamps = new double[512];
            while (true)
            {
                int num = inlet.pull_chunk(buffer,timestamps);
                for (int s = 0; s < num; s++)
                {
                    for (int c = 0; c < 8; c++)
                        Console.Write("\t{0}", buffer[s, c]);
                    Console.WriteLine();
                }
            }
        }
    }
}
